### PR TITLE
fix(iot-svc): Cleanup and deprecation warning of code in CryptoKeyGenerator

### PR DESCRIPTION
**Pull request by drwill-ms**
A service team member pointed out they like to use this method we have `GeneratePassword` for testing, but they are on net472 and it doesn't work for them.

Consequently, the method we call to generate the password, off the `System.Web.Security.Membership` is not available after net472, so even if we do expand support for this method to net472, we can't offer it for newer .net targets unless we find another way to do it.

The service team member was happy to copy the code and use it locally, so this isn't urgent.

> Edit: team decided to not expand support, and mark existing methods as obsolete.

### CODE REVIEW

File: iothub/service/src/Common/Security/CryptoKeyGenerator.cs [hunk 1]
```diff
@@ -67,19 +66,18 @@ public static string GenerateKey(int keySize)
             return Convert.ToBase64String(GenerateKeyBytes(keySize));
         }
 
-#if NET451
+#if NET451 || NET472
```
Comment (drwill-ms) [inline iothub/service/src/Common/Security/CryptoKeyGenerator.cs:69]: This is the key to enabling it for net472.

File: iothub/service/src/Common/Security/CryptoKeyGenerator.cs [hunk 2]
```diff
@@ -38,20 +41,16 @@ static public class CryptoKeyGenerator
         /// <returns>Byte array representing the key.</returns>
         public static byte[] GenerateKeyBytes(int keySize)
         {
-#if !NET451
```
Comment (drwill-ms) [inline iothub/service/src/Common/Security/CryptoKeyGenerator.cs:41]: Flipped the logic so it reads easier.

File: iothub/service/src/Common/Security/CryptoKeyGenerator.cs [hunk 3]
```diff
@@ -38,20 +41,16 @@ static public class CryptoKeyGenerator
         /// <returns>Byte array representing the key.</returns>
         public static byte[] GenerateKeyBytes(int keySize)
```
Comment (jamdavi) [inline iothub/service/src/Common/Security/CryptoKeyGenerator.cs:42]: Did you want to just go ahead and mark this depreciated?

File: iothub/service/src/Common/Security/CryptoKeyGenerator.cs [hunk 4]
```diff
@@ -22,8 +25,8 @@ namespace Microsoft.Azure.Devices.Common
     static public class CryptoKeyGenerator
```
Comment (timtay-microsoft) [inline iothub/service/src/Common/Security/CryptoKeyGenerator.cs:25]: The order of the modifiers here is throwing me off. Can you change this to "public static class" instead?

**Review by timtay-microsoft — APPROVED** [review]

**Review by azabbasi — APPROVED** [review]
It appears that CS0618 is being thrown by the build agent, indicating the obsolete method has references.

**Comment by drwill-ms** [comment]
> It appears that CS0618 is being thrown by the build agent, indicating the obsolete method has references.

I'll make a test helper that isn't marked as obsolete so our test code has a local copy.

**Review by jamdavi — APPROVED** [review]

**Pull request merged**